torch/csrc/jit/tensorexpr/ir.h

```diff
@@ -679,76 +695,15 @@ class Intrinsics : public CallNode<Intrinsics> {
   }
 
   std::string func_name() const override {
-    switch (op_type()) {
-      case kSin:
-        return "sin";
-      case kCos:
-        return "cos";
-      case kTan:
-        return "tan";
-      case kAsin:
-        return "asin";
-      case kAcos:
-        return "acos";
-      case kAtan:
-        return "atan";
-      case kAtan2:
-        return "atan2";
-      case kSinh:
-        return "sinh";
-      case kCosh:
-        return "cosh";
-      case kTanh:
-        return "tanh";
-      case kSigmoid:
-        return "sigmoid";
-      case kExp:
-        return "exp";
-      case kFabs:
-        return "fabs";
-      case kLog:
-        return "log";
-      case kLog2:
-        return "log2";
-      case kLog10:
-        return "log10";
-      case kLog1p:
-        return "log1p";
-      case kErf:
-        return "erf";
-      case kSqrt:
-        return "sqrt";
-      case kRsqrt:
-        return "rsqrt";
-      case kPow:
-        return "pow";
-      case kCeil:
-        return "ceil";
-      case kFloor:
-        return "floor";
-      case kRound:
-        return "round";
-      case kTrunc:
-        return "trunc";
-      case kRand:
-        return "rand";
-      case kFmod:
-        return "fmod";
-      case kRemainder:
-        return "remainder";
-      case kLgamma:
-        return "lgamma";
-      case kExpm1:
-        return "expm1";
-      case kErfc:
-        return "erfc";
-      case kFrac:
-        return "frac";
-      default:
-        throw std::runtime_error(
-            "invalid op_type: " + c10::to_string(op_type()));
-    }
+#define DEFINE_INTRINSIC_CASE(name) \
+  case k##name:                     \
+    return #name;
```

Comment:
I don't think that's correct for, e.g. "kRsqrt".

Reply:
the capitalization is slightly different, is that imporant ?

Reply:
I suspect that the name is what we could be printing in cuda codegen, i.e. it is then compiled and thus the capitalization is important.

Reply:
Yeah thats correct, good catch @ZolotukhinM.